tools/code_format/check_format_go_code.sh

```diff
@@ -13,10 +13,10 @@ cd "$root" || exit 1
 go_files=$(find . -name "*.go")
```

Comment:
arggh - maybe not 

> The problem is that the shell performs word splitting on the results of find so that the elements of the array are not guaranteed to be what you want.

https://stackoverflow.com/questions/23356779/how-can-i-store-the-find-command-results-as-an-array-in-bash

chatGPT says:

```bash

#!/bin/bash

IFS=$'\n'
find /path/to/search -type f -print0 | while read -d $'\0' file; do
    echo "Processing $file"
    # Add any additional commands here
done

```

Reply:
...

```bash

#!/bin/bash

go_files=()

IFS=$'\n'
find /path/to/search -type f -print0 | while read -d $'\0' file; do
    go_files=+("$file")
done

```

Reply:
...

```bash
#!/bin/bash

go_files=()

IFS=$'\n'
while IFS= read -r -d $'\0' file; do
    go_files+=("$file")
done < <(find /path/to/search -type f -print0)


```

Reply:
i think you need this ^^ @doujiang24

Reply:
Oh, I added parentheses. And I tested it locally, it works, even without parentheses.
I think it's good enough for us. We do not use space in filenames.

Maybe we can keep it simple? I'm fine to change it if you think `while read` could be better.

Reply:
> I think it's good enough for us. 

lets do it the right way

Reply:
Thanks, already fixed it, also tested the shell works on my local side.